x-pack/plugins/infra/server/lib/adapters/nodes/lib/create_node_item.ts

```diff
@@ -4,18 +4,46 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
+import { last } from 'lodash';
+import { isNumber } from 'lodash';
+import moment from 'moment';
 import { InfraNode, InfraNodeMetric } from '../../../../../common/graphql/types';
 import { InfraBucket, InfraNodeRequestOptions } from '../adapter_types';
+import { getBucketSizeInSeconds } from './get_bucket_size_in_seconds';
 
 // TODO: Break these function into seperate files and expand beyond just documnet count
 // In the code below it looks like overkill to split these three functions out
 // but in reality the create metrics functions will be different per node type.
+
+const findLastFullBucket = (
+  bucket: InfraBucket,
+  bucketSize: number,
+  options: InfraNodeRequestOptions
+) => {
+  const { buckets } = bucket.timeseries;
+  const to = moment.utc(options.timerange.to);
+  return buckets.reduce((current, item) => {
+    const itemKey = isNumber(item.key) ? item.key : parseInt(item.key, 10);
+    const date = moment.utc(itemKey + bucketSize * 1000);
+    if (!date.isAfter(to) && item.doc_count > 0) {
+      return item;
+    }
+    return current;
+  }, last(buckets));
+};
+
 function createNodeMetrics(
   options: InfraNodeRequestOptions,
   node: InfraBucket,
   bucket: InfraBucket
 ): InfraNodeMetric[] {
-  return [{ name: 'count', value: bucket.doc_count }];
+  const { timerange, metrics } = options;
+  const bucketSize = getBucketSizeInSeconds(timerange.interval);
+  const lastBucket = findLastFullBucket(bucket, bucketSize, options);
```

Comment:
What about the case that `bucket.timeseries.buckets` is an empty array and `lastBucket` therefore `undefined`?

Reply:
I'm not sure this would ever be null or empty unless the entire request failed (which would throw at the callWithRequest level). I added an exception if it happens to be empty.

Reply:
True, but the code path leading here could change without that being obvious. So a check would be nice :+1: